### DISCUSSION TITLE
how to do _in filter in many to many relation field

### DISCUSSION THREAD

**Discussion by sambitraze** [directus/directus#12203] [Archive]
I have a college collection and it has a field of Many to Many relations called college_for_types(college type collection) but I am not able to do the filtering according to college type. I want to fetch college which contains one or two of the college types. I tried with deep but it filtered the data inside the related field, not the college itself.

`https://api.collegeking.in/items/college?filter={"college_for_types":{"_in":[3]}}` this is the command i am supoose to use but its nto working it say there is no field called college_for_types.in

`{
"collection": "college",
"field": "college_for_types",
"type": "alias",
"schema": null,
["meta"](https://api.collegeking.in/fields/college): {
"id": 58,
"collection": "college",
"field": "college_for_types",
["special"](https://api.collegeking.in/fields/college): [
"m2m"
],
"interface": "list-m2m",
["options"](https://api.collegeking.in/fields/college): {
"template": "{{CollegeFor_id.title}}"
},
"display": "related-values",
["display_options"](https://api.collegeking.in/fields/college): {
"template": "{{CollegeFor_id.title}}"
},
"readonly": false,
"hidden": false,
"sort": 14,
"width": "full",
"translations": null,
"note": null,
"conditions": null,
"required": true,
"group": null
}
}`

**Comment by azrikahar** [comment]
It should be `filter={"college_for_types":{"id":{"_in":[3]}}}`, since we are checking the `id` 👍 Here's an example:

### Full payload without filter

![image](https://user-images.githubusercontent.com/42867097/158806667-1d60a002-8aa0-458b-9278-15d2be61158c.png)


### Using the above filter

Filter used: `filter={"college_for_types":{"id":{"_in":[3]}}}`

Formatted json of the filter used for clarity:

```json
{
 "college_for_types": {
   "id": {
     "_in": [
       3
     ]
   }
 }
}
```

Result:
    
![image](https://user-images.githubusercontent.com/42867097/158807111-0ec2ce5b-1bf4-4672-bc26-405546bd7381.png)
    
### Alternative syntax

Filter used: `filter[college_for_types][id][_in]=3`

Result:

![image](https://user-images.githubusercontent.com/42867097/158807196-5c027755-c4d6-4f5d-86ef-d1185581497d.png)
